cellomics_mongo.py

```diff
@@ -0,0 +1,204 @@
+
+# coding: utf-8
+
+# In[50]:
+
+# dependencies
+import os
+import pandas as pd
+import sys
+from pymongo import MongoClient
+from bson.binary import Binary
+from skimage import io
+from skimage.transform import resize
+from microscopium.screens import cellomics
+from sklearn.decomposition import PCA
+from sklearn.neighbors import NearestNeighbors
+from sklearn.preprocessing import StandardScaler
+import numpy as np
+
+
+# In[51]:
+
+screen_name = 'LNC-SP-10A'
+
+# load the feature data frame
+def string2tuple(string_tuple):
+    string_values = string_tuple.split(', ')
+    coords = (int(string_values[0][1:]), string_values[1][1:-2])
+    return coords
+
+def new_tuple(coord):
+    return '-'.join((screen_name, str(coord[0]), str(coord[1])))
+
+feature_data = pd.read_csv('10A-LNC-SP-stitch2.csv', index_col=0, converters={0: string2tuple})
+feature_data.head()
+
+# update the keys to use our new format
+old_index = feature_data.index
+new_index = map(new_tuple, feature_data.index)
+feature_data.index = new_index
+feature_data.head()
+
+
+# In[52]:
+
+# normalise data frame
+feature_data_std = StandardScaler().fit_transform(feature_data.values)
+feature_data_std = pd.DataFrame(feature_data_std, index=feature_data.index, columns=feature_data.columns)
+feature_data_std.head()
+
+
+# In[53]:
+
+# need PCA co-ordinates now
+pca = PCA(n_components=2)
+pca_data = pca.fit_transform(feature_data_std)
+pca_data = pd.DataFrame(pca_data, index=new_index)
+pca_data.head()
+
+
+# In[57]:
+
+# now a nearest neighbours dictionary
+neighbours = NearestNeighbors(n_neighbors=25).fit(feature_data_std)
+neighbours_dict = {}
+for sample in feature_data.index:
+    sample_point = feature_data_std.loc[sample].values
+    distances, indices = neighbours.kneighbors(sample_point)
+    neighbours_dict[sample] = list(feature_data.index[indices[0]])
+    
+print neighbours_dict['LNC-SP-10A-140206180002-A01']
```

Comment:
Time to switch to Python 3! =)

For this script, a comment at the top specifying Python 2.7 will suffice.